herddb-core/src/main/java/herddb/core/DBManager.java

```diff
@@ -903,7 +905,18 @@ public void dumpTableSpace(String tableSpace, String dumpId, Pdu message, Channe
             LOGGER.log(Level.SEVERE, "error on dump", error);
         }
     }
-
+      
+    public StatementExecutionResult createTableDigest(TableIntegrityCheckStatement tableIntegrityCheckStatement ){
+        TableSpaceManager manager= tablesSpaces.get(tableIntegrityCheckStatement.getTableSpace());
+        String table = tableIntegrityCheckStatement.getTable();
+        try {   
+            manager.createAndWriteTableDigest(manager,tableIntegrityCheckStatement.getTableSpace(), table);
```

Comment:
This function can return all of the computed data, then you can return that data in a DataIntegrityStatementResult